Functionele documentatie 'zaaktypen' API (#451)
VNG-Realisatie/gemma-zaken: docs/content/functioneel/zaaktypen_api.md

ZAAKTYPEN API
===============================================
**FUNCTIONELE DOCUMENTATIE**

| **Aspect**      | **Beschrijving**                                                                                                                                                                                  |
|-----------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Doel            | Het kunnen aanspreken van een voorziening voor het onderhouden en raadplegen van zaaktypen en de zaaktypecatalogi (ZTC) waarvan zij deel uit maken. |

Reviewer: Ik denk toch dat 'onderhouden' niet het juiste woord is. Dat klinkt al functionaliteit voor het onderhouden terwijl het alleen om het vastleggen van gegevens gaat. Voorstel: '... voor het vastleggen en raadplegen van (gegevens van) zaaktypen ...'.

Reviewer 2: Op dit moment is de API ook enkel-lezen, dus ik zou het zelfs durven beperken tot '... voor het raadplegen van zaaktypen...`

Reviewer 2: @ArjanKloosterboer vertical slicing! ;) Ik denk dat het belangrijk is dat we de functionele documentatie in sync houden met de huidige stand van zaken. Dat dat muteren er moet komen klopt 100%, maar daar moet qua techniek wel even het een en ander voor gebeuren _en_ er moet een userstory voor komen.

Momenteel kan er gemuteerd worden in de admin-interface van het ZTC, buiten de API om.

Reviewer: Ik stel voor de functionele documentatie (ook) als ontwerp te beschouwen en, indien van toepassing, daarin duidelijk te maken welke delen van het ontwerp nog niet in de API gerealiseerd zijn. Anders levert die documentatie direct vragen op: 'waarom kan dat niet'?

Reviewer 3: Hmmm, zo worden er via functionele documentatie opeens functies of constructies bij bedacht die niet persé uit User Stories komen. Zorgt ook voor meer inconsistentie. Dat is een niet wenselijke situatie en gaat in tegen het proces en de DoD.

Reviewer: Dan is er bij het uitwerken en realiseren van de userstory's toch iets mis gegaan. Je kunt alleen gegevens raadplegen als er functionaliteit is om deze te beheren. Dus in de userstory's waarin zaaktypegegevens geraadpleegd worden, had ook muteerfunctionaliteit gebouwd moeten worden op die gegevens. 
@ehotting, hoe kijk jij hier tegen aan als productowner?

Author: Ik ben het wel eens met Sergei: de muteerfunctionaliteit moet ingebouwd worden op basis van een eigen user story. Anders heeft m.i. het werken volgens userstories geen zin en kun je beter een complete REST API op alle resources in het RGBZ bouwen en achter kijken of alle userstories met die API in te vullen zijn.
Precies wat Joeri dan aangeeft gaat dat tegen het proces en de DoD in.

Reviewer 2: Ik sluit aan bij Joeri en Michiel - als je alles op voorhand gaat uitdenken en neerschrijven in functionele documentatie, dan:

* gaat het heel erg op verkapte waterval lijken
* is de functionele documentatie helemaal niet in sync met de functionaliteit

Ik dacht dat het de bedoeling was om functionele documentatie samen met API spec en referentieimplementatie op basis van een user story bij te werken, en dat is een geheel, 1 slice van functionaliteit.

Reviewer: Zit wat in. Maar, volgens mij ontbreken de muteer-operaties. 

> Je kunt alleen gegevens raadplegen als er functionaliteit is om deze er in te stoppen. Dus in de userstory's waarin zaaktypegegevens geraadpleegd worden, had ook muteerfunctionaliteit gebouwd moeten worden op die gegevens.

Author: Dan moet daar een userstory voor komen. De documentatie beschrijft wat er nu is.

| Domein          | Zaakgericht werken                                                                                                                                                                                |
| Provider        | [Zaaktypecataloguscomponent](https://www.gemmaonline.nl/index.php/GEMMA2/0.9/id-3ef9cdd9-631c-4d3e-88c3-f756423d6314) (GEMMA2)                                                              |
| Consumer        | Componenten waarmee zaaktypen beheerd worden en componenten die zaaktypegegevens gebruiken voor het behandelen van zaken, zoals Zaakafhandelcomponenten (ZAC’s).                                                              |

> | Informatiemodel | [IMZTC, versie 2.2](https://www.gemmaonline.nl/images/gemmaonline/1/1d/EgIM_20180621_-_ag_4_-_GEMMA_ImZTC_2.2_-_CONCEPT_20180613.zip) (in-ontwikkeling)                                         |

Reviewer 2: Muggenzifterij: is het niet `ImZTC` in plaats van `IMZTC`?

Reviewer: Scherp!

Reviewer 2: Haha, ik hoop dat dat niet voldoende competenties zijn om tot kenner gebombardeerd te worden, anders ga ik me zorgen maken :smiley:

Reviewer 4: en andersom..
ik weet ook dat het ImZTC is...:-)

> | Specificaties   | <https://ref.tst.vng.cloud/zrc/api/v1/schema>                                                                                                                                                     |

Reviewer 2: `zrc` in link moet `ztc` zijn

Author: beide commentaren verwerkt!

| Bijzonderheden  | -                                                    |

> Deze API omvat (vooralsnog) de resources Catalogussen, Zaaktypen, Informatieobjecttypen, Roltypen, Statustypen en Zaakobjecten.

Reviewer 2: klopt niet, resources zijn: Catalogus, Informatieobjecttype, Zaaktype, Eigenschap, Roltype en Statustype. Dus geen Zaakobjecten.

Author: tikfoutje, dat had moeten Besluittype moeten zijn.

> Op dit moment bieden de resources alleen de mogelijkheid tot raadplegen van de gegevens (GET).

Reviewer 2: De resources bieden niets, de API daarentegen 'laat op dit moment enkel raadplegen van de resources toe'.

Reviewer: Je raadpleegt toch een database, geen resource? Een resource maakt dat mogelijk, toch?

Reviewer 2: Als ik https://restful-api-design.readthedocs.io/en/latest/resources.html nalees:

> The fundamental concept in any RESTful API is the resource. A resource is an object with a type, associated data, relationships to other resources, and a set of methods that operate on it. It is similar to an object instance in an object-oriented programming language, with the important difference that only a few standard methods are defined for the resource (corresponding to the standard HTTP GET, POST, PUT and DELETE methods), while an object instance typically has many methods.

Dan kan de originele verwoording inderdaad blijven staan.

> ### Resource: catalogus

Reviewer: De naam van de resource is toch in meervoud? Geldt ook voor de hierna volgende resources.

Author: Alle resources zijn inmiddels meervoud

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Aanleiding     | Een consumer wil een lijst van alle beschikbare CATALOGUSsen of een specifieke catalogus opvragen.         |
| *Toevoegen*    |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Wijzigen*     |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Verwijderen*  |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Raadplegen*   |      |
| - voorwaarde   | nvt.         |
| - gevolg       | Een lijst met beschikbare catalogussen wordt teruggegeven aan de consumer bij het opvragen van de lijst.<br>Een specificieke catalogus wordt teruggegeven aan de consumer. |

> | Gegevens       | objecttype CATALOGUS inclusief een lijst met relaties naar daarin gedefinieerde ZAAKTYPEn.<br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Author: Ik denk dat de omschrijving "raadplegen" te breed is. Een ZTC kan verschillende, meer dan één, catalogussen bevatten.

Raadplegen van "/catalogussen"  geeft een lijst terug van alle catalogussen in de geraadpleegde ZTC. 
Raadplegen van "/catalogussen/{UUID}" geeft een specifieke catalogus terug, inclusief de inhoud (relaties naar zaaktypen).

Vergelijk het het met het opvragen van zaken. Wanneer ik een lijst van zaken opvraag krijg ik alleen de URIs terug. Vraag ik een zaak met één van die URI sop krijg ik alle gegevens van die zaak, inclusief URI's (relaties) naar eventuele zaakinformatieobjecten, statussen enz.

> | Gegevens       | objecttype CATALOGUS inclusief een lijst met relaties naar daarin gedefinieerde ZAAKTYPEn.<br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Reviewer: Mijn opmerking sloeg op de tekst achter 'Gegevens', jouw reactie lijkt ergens anders over te gaan. Tekstvoorstel: "objecttype CATALOGUS inclusief relatie naar ZAAKTYPEn.<br>Zie de volgende tabel voor de gegevens van deze resource." 
En ik ben benieuwd naar de reactie op de tweede alinea van mijn comment (die over de relatie zaaktype - catalogus). @sergei-maertens?

> | Gegevens       | objecttype CATALOGUS inclusief een lijst met relaties naar daarin gedefinieerde ZAAKTYPEn.<br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Author: Je opmerking over de relatie is een bevinding op de OAS. De documentatie beschrijft de OAS dus dat is goed. Gebruik de review van de documentatie svp niet om de OAS te reviewen, dat moet in een apart issue/pull-request.

> | Gegevens       | objecttype CATALOGUS inclusief een lijst met relaties naar daarin gedefinieerde ZAAKTYPEn.<br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Reviewer 2: @ArjanKloosterboer 

> Wat me verbaast is dat dat de relatie naar zaakype deel uit maakt van deze resource. Ik zou die precies omgekeerd verwachten. Wat is de reden voor deze (technische) keuze en in welke userstory is die gemaakt?

ZTC vertrekpunt was ZTC dat voor Haarlem gebouwd is/was, en daar was blijkbaar behoefte aan. Bij mijn weten is er geen userstory voor. De relatie is gewoon in beide richtingen - zaaktype geeft aan bij welke catalogus het hoort en catalogus lijst alle beschikbare zaaktypen op.

> | Specificaties  | https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/catalogus                                                                                                   |

Reviewer 2: https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/catalogussen lijkt de correcte URL te zijn

Author: Maar ik zie geen schema van het objecttype catalogus, alleen van de response bij een 200. Is dat dan hetzelfde schema als gebruikt wordt wanneer ik (later, is nu nog niet ingebouwd) een catalogus toe zou voegen via een POST?

Reviewer 2: Jep. REST - state van een resource, je gooit dus altijd dezelfde resource over de lijn in beide richtingen (kort door de bocht, maar bijna correct).

Author: Dat snap ik wel maar het gaat er om dat bij het beschrijven van een resource naar een operatie verwezen wordt. Terwijl die resource straks bij verschillende operaties van toepassing is. 
Om de verwijzing (in de documentatie) zuiver te houden moet in de documentatie verwezen worden naar een schema van de betrokken resource, niet naar een operatie waar toevallig ook een vergelijkbaar schema in gebruikt wordt.

Een andere mogelijkheid is per operatie een verwijzing te maken, alleen durf ik niet te garanderen dat het er dan duidelijker op wordt

Author: In de Zaken API documentatie heb ik bij wijze van experiment een verwijzing per operatie toegevoegd. Persoonlijk vind ik dat duidelijker.

| Samenhang      | nvt. |
| Bijzonderheden | nvt. |
        

### Resource: zaaktype

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aanleiding     | Een consumer wil (een lijst van beschikbare) zaaktype informatie opvragen uit de catalogus.            |

Author: verwerkt

| *Toevoegen*    |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Wijzigen*     |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Verwijderen*  |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Raadplegen*   |      |

> | - voorwaarde   | Om zaaktype informatie op te kunnen vragen moet de consumer de UUID van de catalogus waarin het zaaktype is opgenomen weten. Deze UUID kan opgevraagd worden met de catalogus bevragingen services.       |

Reviewer 2: Ik zou eerder stellen dat de lijst van zaaktypen uit de list/detail van catalogus gebruikt moet worden. Het zelf ineenpuzzelen van de URLs met de parameters zou ik niet aanmoedigen.

Author: @ArjanKloosterboer  je slaat de spijker op zijn kop: ik ben een techneut :-)

Author: > 
> 
> Ik zou eerder stellen dat de lijst van zaaktypen uit de list/detail van catalogus gebruikt moet worden. Het zelf ineenpuzzelen van de URLs met de parameters zou ik niet aanmoedigen.

Dit was niet bedoeld om zelf URLs in elkaar te knutselen maar meer als discovery service: eerst vragen aan de ZTC component welke catalogussen er zijn, dan de zaaktypen bij de juiste catalogus opvragen.

Author: Tekst is inmiddels aangepast.

> | - gevolg       | Een lijst met in de catalogus opgenomen zaaktypen wordt teruggegeven aan de consumer bij het opvragen van de lijst.<br>Een specificiek zaaktype wordt teruggegeven aan de consumer.         |

Author: Ja, tenminste, afhankelijk van wat je opvraagt. Alleen zijn dat beide varianten van "Raadplegen". En je kunt zelfs een hele hoop varianten van raadplegen maken, als een soort voorgedefinieerde queries. 
Moeten we die allemaal onder hetzelfde kopje "Raadplegen" beschrijven of is het beter om die apart te benoemen. Het zijn immers aparte operaties.

> | - gevolg       | Een lijst met in de catalogus opgenomen zaaktypen wordt teruggegeven aan de consumer bij het opvragen van de lijst.<br>Een specificiek zaaktype wordt teruggegeven aan de consumer.         |

Author: Ik heb ervoor gekozen de varianten apart te benoemen om redenen die ik elders in een commentaar opgenoemd heb.

> | Gegevens       | ZAAKTYPE inclusief STATUSTYPEn, EIGENSCHAPpen en ROLTYPEn<br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Author: Dit is een bevinding op de OAS. De documentatie beschrijft de OAS dus dat is goed. Gebruik de review van de documentatie svp niet om de OAS te reviewen, dat moet in een apart issue/pull-request.

> | Gegevens       | ZAAKTYPE inclusief STATUSTYPEn, EIGENSCHAPpen en ROLTYPEn<br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Author: > Ik zie geen wijziging en ook geen motivering waarom dat niet is gedaan.
>  En graag beginnen met 'Objecttype ...'.

De tekst is volledig anders, mogelijk kijk je naar een oude versie? De tekst die in het groene blok staat inmiddels net meer in de documentatie. Daarom resolve ik deze conversatie.

> | Specificaties  | https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/zaaktype                                                                                |

Reviewer 2: deze URL werkt niet

Author: Dat was mij ook al opgevallen, hoe krijgen we dit wel werken? In Redoc kan ik veel dingen niet die ik met swaggerhub wel kan. Dat kan mijn gebrek aan kennis zijn of mist er iets aan de OAS?

Reviewer 3: Heeft niets met de OAS te maken (Redoc of Swaggerhub zijn maar grafische layouts om dezelfde OAS). Wel met de prioriteiten van Redoc features, waar dit punt laag op de agenda staat. 

Er vragen meer mensen om, zie: https://github.com/Rebilly/ReDoc/issues/134

Author: Ok, voor nu is het wel belangrijk dat we dit oplossen/omzeilen. VNG heeft een Swaggerhub omgeving beschikbaar, kunnen we daar dingen wel die niet met Redoc kunnen? Dan stel ik voor de best geschikte tooling te gebruiken en ons niet te laten beperken omdat een leverancier iets niet ingebouwd heeft :-)

Reviewer 3: Maak er een User Story voor aan :)

> Dan stel ik voor de best geschikte tooling te gebruiken en ons niet te laten beperken omdat een leverancier iets niet ingebouwd heeft :-)

Uiteraard willen we de best geschikte tool. Maar voor wie? Het is een kwestie van rol/perspectief.

Voor mij voldoet bijvoorbeeld: https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/zaaktype_read

> VNG heeft een Swaggerhub omgeving beschikbaar, kunnen we daar dingen wel die niet met Redoc kunnen

Ik merk enige drang om naar een betaalde omgeving te switchen. Dat is niet nodig. SwaggerUI is gewoon een beschikbaar schilletje om een OAS. We kunnen het zo aanzetten dat beide beschikbaar zijn.

Dus, why not both?

Reviewer: > Voor mij voldoet bijvoorbeeld: https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/zaaktype_read

Voor mij ook. Dit is 'm, in dit geval.

Author: > 
> > Dan stel ik voor de best geschikte tooling te gebruiken en ons niet te laten beperken omdat een leverancier iets niet ingebouwd heeft :-)
> 
> Uiteraard willen we de best geschikte tool. Maar voor wie? Het is een kwestie van rol/perspectief.

Dat is wat mij betreft de tool die het werk van ieder het beste en op de eenvoudigste manier ondersteunt.

> 
> Voor mij voldoet bijvoorbeeld: https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/zaaktype_read

Ziet er goed uit, is er ergens een beschrijving hoe ik deze url's op kan vragen? Ik denk nu wel te snappen hoe ik ze zelf kan afleiden maar als er ergens een slimme truc is om het gegarandeerd goed te doen leer ik die graag.

> 
> > VNG heeft een Swaggerhub omgeving beschikbaar, kunnen we daar dingen wel die niet met Redoc kunnen
> 
> Ik merk enige drang om naar een betaalde omgeving te switchen. Dat is niet nodig. SwaggerUI is gewoon een beschikbaar schilletje om een OAS. We kunnen het zo aanzetten dat beide beschikbaar zijn.

Het is geen drang, meer gebrek aan kennis en de beschikking hebben over een tool die dat gebrek lijkt te kunnen compenseren.

> 
> Dus, why not both?

Vind ik prima, mag van mij ook gerust alleen redoc zijn alleen wil dan graag alles kunnen wat ik met swaggerhub kan en weten hoe dat moet (vooral dat laatste vermoed ik).

Author: > 
> 
> > Voor mij voldoet bijvoorbeeld: https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/zaaktype_read
> 
> Voor mij ook. Dit is 'm, in dit geval.

Eigenlijk is dit hem niet: de Resource Zaaktype lijkt nu beschreven in het kader van de lees-operatie. Terwijl het schema van Zaaktype uiteindelijk ook voor (de nog toe te voegen) muteer-operaties zal gaan gelden (verwacht ik). Ik zou hier dus een link naar het schema van het objecttype verwachten en niet naar de operatie zaaktype_read.

Reviewer: Het zou m.i. een link moeten zijn naar de resource (zoiets als https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/zaaktypen) maar die resource lijkt niet onderscheiden te worden.

Author: precies wat ik bedoel!

| Samenhang      | - |
| Bijzonderheden | -           |

> ### Resource: BESLUITTYPE

Reviewer 2: Bestaat nog niet in master-versie van ZTC (komt er wel aan binnenkort). Dus even `TODO` bijzetten

Author: Besluittype bestaat al wel in de master versie, https://github.com/VNG-Realisatie/gemma-zaken/blob/master/api-specificatie/ztc/openapi.yaml op regel 1298.
Anders had ik het niet in de documentatie opgenomen :-D

Reviewer 2: Lomp van mij :-)

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aanleiding     | Een consumer wil alle BESLUITTYPEn van de besluiten die het resultaat kunnen zijn van het zaakgericht werken van de behandelende organisatie(s) opvragen.            |

Author: check

| *Toevoegen*    |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Wijzigen*     |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Verwijderen*  |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Raadplegen*   |      |
| - voorwaarde   | Om besluittype informatie op te kunnen vragen moet de consumer de UUID van de catalogus waarin het besluittype is opgenomen weten. Deze UUID kan opgevraagd worden met de catalogus bevragingen services.
         |

> | - gevolg       | Een (lijst met) besluittype informatie zoals opgenomen in de catalogus wordt teruggegeven.             |

Author: is aangepast

| Gegevens       | Besluittype <br>Zie de volgende tabel voor de gegevens van deze resource                                 |

> | Specificaties  | [https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/besluittype                                                                                                   |

Reviewer 2: deze zal het ook niet doen

Author: Ook hiervoor geldt dezelfde vraag: hoe kom ik achter de correcte url's?

| Samenhang      | - |

> | Bijzonderheden | -           |

Author: En die reden is?

> ### Resource: INFORMATIEOBJECTTYPE

Author: is aangepast

| *Toevoegen*    |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Wijzigen*     |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Verwijderen*  |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Raadplegen*   |      |
| - voorwaarde   | Om informatieobjecttype informatie op te kunnen vragen moet de consumer de UUID van de catalogus waarin het informatieobjecttype is opgenomen weten. Deze UUID kan opgevraagd worden met de catalogus bevragingen services.
         |
| - gevolg       | Een (lijst met) besluittype informatie zoals opgenomen in de catalogus wordt teruggegeven.             |
| Gegevens       | Informatieobjecttype <br>Zie de volgende tabel voor de gegevens van deze resource                                 |
| Specificaties  | [https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/informatieobjecttype                                                                                                   |
| Samenhang      | - |
| Bijzonderheden | -           |

> ### Resource: eigenschap

Author: Tekst inmiddels aangepast.

| - voorwaarde   | nvt.         |
| - gevolg       | nvt.            |
| *Wijzigen*     |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Verwijderen*  |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Raadplegen*   |      |
| - voorwaarde   | Om EIGENSCHAP informatie op te kunnen vragen moet de consumer de UUID van de CATALOGUS en de UUID van het ZAAKTYPE waarin de EIGENSCHAP is opgenomen weten. Deze UUIDs kunnen opgevraagd worden met de catalogus bevragingen services en ZAAKTYPE bevragingen services.         |
| - gevolg       | Een (lijst met) eigenschap informatie van het ZAAKTYPE wordt teruggegeven.            |
| Gegevens       | EIGENSCHAP <br>Zie de volgende tabel voor de gegevens van deze resource                                 |
| Specificaties  | [https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/eigenschap                                                                                                   |
| Samenhang      | Een relevant inhoudelijk gegeven dat bij ZAAKen van dit ZAAKTYPE geregistreerd moet kunnen worden en geen standaard kenmerk is van een zaak.  |
| Bijzonderheden |      |

      

> ### Resource: roltype

Reviewer: Daar houd ik het maar even bij want ik val in herhalingen.

| *Toevoegen*    |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Wijzigen*     |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Verwijderen*  |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |
| *Raadplegen*   |      |
| - voorwaarde   | Om ROLTYPE informatie op te kunnen vragen moet de consumer de UUID van de CATALOGUS en de UUID van het ZAAKTYPE waarin het ROLTYPE is opgenomen weten. Deze UUIDs kunnen opgevraagd worden met de catalogus bevragingen services en zaaktype bevragingen services.         |
| - gevolg       | Een (lijst met) ROLTYPE informatie van het ZAAKTYPE wordt teruggegeven aan de consumer.             |
| Gegevens       | ROLTYPE <br>Zie de volgende tabel voor de gegevens van deze resource                                 |
| Specificaties  | https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/roltype                                                                                                   |
| Samenhang      | - |
| Bijzonderheden | Generieke aanduiding van de aard van een ROL die een BETROKKENE kan uitoefenen in ZAAKen van een ZAAKTYPE.       |

       

> ### Resource: STATUSTYPE

Reviewer: Daar houd ik het maar even bij want ik val in herhalingen.

ZAAKTYPEN API
===============================================
**FUNCTIONELE DOCUMENTATIE**

[aanzet tot documentatie van de API’s van het ZDS2-koppelvlak. Is vooralsnog een
‘try-out’ n.a.v. [userstory
\#188](https://github.com/VNG-Realisatie/gemma-zaken/issues/188) om inzicht te
krijgen in de wijze van (functioneel) documenteren van API’s, als onderdeel van

> koppelvlakdocumentatie]

Reviewer: Ik heb deze alinea verwijderd, kan er nu wel uit, m.i.

| **Aspect**      | **Beschrijving**                                                                                                                                                                                  |
|-----------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Doel            | Het kunnen aanspreken van een voorziening voor het onderhouden en raadplegen van zaaktypecatalogus (ZTC) inclusief daarinopgenomen zaaktypen, statustypen, roltypen en informatieobjecttypen, rollen en relaties naar betrokkenen, objecten, zaaktypen en statustypen. |

Reviewer: Ik heb het accent op de zaaktypen gelegd, omdat het m.i. daar om draait. En de details weggelaten. En van het laatste deel ('relaties naar ..') is het maar de vraag of dat terecht is, lijkt me van niet.

| Domein          | Zaakgericht werken                                                                                                                                                                                |

> | Provider        | [Zaaktypecataloguscomponent](https://www.gemmaonline.nl/index.php/Omgevingswet/1.5/id-3ef9cdd9-631c-4d3e-88c3-f756423d6314) (GEMMA2)                                                              |

Reviewer: Volgens mij link je naar de verkeerde component (die in het Omgevingewetdomein) en moet het die naar GEMMA zijn. Heb ik aangepast.

> | Consumer        | Componenten waarmee zaken behandeld worden (Zaakafhandelcomponenten; ZAC’s) en (andere) componenten die zaaktypegegevens raadplegen.                                                              |

Reviewer: Ik heb dit aangevuld en aangescherpt (beheer van zaaktypen ontbrak).

> | Informatiemodel | [IMZTC, versie 2.1](https://www.gemmaonline.nl/index.php/Informatiemodel_Zaaktypen_(ImZTC)) (in-ontwikkeling)                                         |

Reviewer: We zijn al weer verder: 2.2. Heb ik aangepast, ook de link.

| Specificaties   | <https://ref.tst.vng.cloud/zrc/api/v1/schema>                                                                                                                                                     |

> | Bijzonderheden  |  Deze API omvat (vooralsnog) de resources Catalogussen, Zaaktypen, Informatieobjecttypen, Roltypen, Statustypen en Zaakobjecten                                                    |

Reviewer: Deze tekst hoort een regel lager waar al een vergelijkbare maar onjuiste tekst staat. Ik heb 'm daarnaar verplaatst.

> Deze API omvat (vooralsnog) de resources Zaken, Klantcontacten, Rollen, Statussen en Eigenschappen.

Reviewer: Ik heb dit vervangen door de tekst die bij 'Bijzonderheden' stond. Maar ook die klopt m.i. niet. Check 'm svp met de API-specificatie.

> Op dit moment biedt deze API alleen nog de mogelijkheid de zaaktypecatalogus te raadplegen middels GET.

Reviewer: Ik heb de tekst aangepast omdat het hier over de resources zou moeten gaan, niet over de API als geheel. En ik vermijd daarmee de term 'zaaktypecatalogus' (want het draait om de zaaktypen).

### Resource: catalogus

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Aanleiding     | Een consumer wil een lijst van alle beschikbare CATALOGUSsen of een specifieke catalogus opvragen.         |
| *Toevoegen*    |      |
| - voorwaarde   | nvt.         |

> | - gevolg       | nvt.             |

Reviewer: Ik zou bij 'Toevoegen', omdat deze operatie nog niet ontwikkeld is, zeggen '(operatie nog niet beschikbaar)'.  En dan bij voorwaarde en gevolg 'n.t.b.' ipv. 'n.v.t.'. Ze zijn wel degelijk van toepassing maar we kennen ze nog niet. In de andere specificaties hieronder naar analogie.

Reviewer 2: Wat moet er dan nog toegevoegd worden? De operaties `catalogus_list` en `catalogus_read` zijn er al.

Reviewer: Post! (spreek uit: poost).

Author: tekst aangepast.

| *Wijzigen*     |      |
| - voorwaarde   | nvt.         |
| - gevolg       | nvt.             |

> | *Verwijderen*  |      |

Reviewer: Catalogi kunnen net als Zaaktypen niet verwijderd worden. dit is dus echt 'n.v.t. (een Catalogus kan niet verwijderd worden, alleen beeindigd).' Dan kun je de rijen 'voorwaarde' en 'gevolg' weglaten.

Reviewer 2: Is `Niet toegestaan` dan niet duidelijker om verwarring te voorkomen?

Author: Als we ZTC informatie willen beheren (toevoegen, bijwerken) vind ik eigenlijk dat we ook moeten kunnen verwijderen. Dit mag dan weliswaar alleen door de juiste persoon (beheerder oid) gedaan worden maar dat moet dan afgevangen worden met de juiste autorisatie en validaties. Datzelfde geldt voor zaken en documenten. Of komt er een aparte API voor het beheer van de gegevens?

Ik denk dan even aan het geval dat bijvoorbeeld een zaak vernietigd moet worden. Op één of andere wijze moet dit gecontroleerd kunnen gebeuren. Moet dit dan heel low-level op de database gedaan worden? Voor mijn gevoel moet er een API zijn om in zo'n geval te kunnen verwijderen.

De ontwerpkeuze "dat er van verwijderen geen sprake kan zijn" is naar mijn mening te kort door de bocht en niet correct. Een gewone zaakgerichte consumer als een generieke of specifieke Zaakafhandelcomponent (ZAC) kan niet verwijderen maar een archiefregistratiecomponent oid zou dat wel moeten kunnen.

Reviewer: Scherp! Uiteindelijk moet alles verwijderd kunnen worden maar dan onder zeer geconditioneerde omstandigheden. Zo mag een zaaktype pas fysiek verwijderd worden als alle daarop gebaseerde zaken vanwege archiefregime zijn vernietigd dan wel 'overgebracht'. Ik stel voor om het verwijderen van wat dan ook vooralsnog buiten scope te plaatsen. Eerst maar 's een userstory die daar over gaat.

Reviewer 2: Heren, dit is een nuttige discussie die in een eigen user-story gevoerd moet worden :-) De scope van deze functionele documentatie is op dit moment al breed genoeg, het lijkt me handig om te focussen op de huidige stand van zaken.

Author: Dan pakken we dit op bij een userstory en sluit ik deze conversatie

| Informatiemodel | [ImZTC, versie 2.2](https://www.gemmaonline.nl/images/gemmaonline/1/1d/EgIM_20180621_-_ag_4_-_GEMMA_ImZTC_2.2_-_CONCEPT_20180613.zip) (in-ontwikkeling)                                         |
| Specificaties   | <https://ref.tst.vng.cloud/ztc/api/v1/schema>        | 
| Bijzonderheden  | -                                                    |

Deze API omvat (vooralsnog) de resources Catalogussen, Zaaktypen, Informatieobjecttypen, Roltypen, Statustypen en Besluittypen.

Op dit moment bieden de resources alleen de mogelijkheid tot raadplegen van de gegevens (GET).

### Resource: catalogus

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aanleiding     | Een consumer wil informatie van alle beschikbare catalogussen of een specifieke catalogus opvragen.         |

Author: Het is inderdaad wel een krommige zin.

> | Aanleiding     | Een consumer wil informatie van alle beschikbare catalogussen of een specifieke catalogus opvragen.         |

Author: Aangepast.

| *Toevoegen*    |      |
| - voorwaarde   | Nog niet beschikbaar.         |

> | - gevolg       | ntb.             |

Reviewer 2: Ik ben zelf wel fan van voluit schrijven in plaats van afkortingen :wink:

| *Wijzigen*     |      |
| - voorwaarde   | Nog niet beschikbaar.         |
| - gevolg       | Nog te bepalen.             |
| *Verwijderen*  |      |
| - voorwaarde   | Nog niet beschikbaar.         |
| - gevolg       | Nog te bepalen.             |
| *Raadplegen lijst*   |      |
| - voorwaarde   | Geen aanvullende voorwaarden gesteld.  |
| - gevolg       | Een lijst met beschikbare catalogussen wordt teruggegeven aan de consumer bij het opvragen van de lijst. |

> | *Raadplegen details*   |      |

Author: Ik maak dat onderscheid bewust omdat:
1/ er worden andere eisen gesteld aan de operaties
2/ er komen andere antwoorden terug van de operaties

Om het overzichtelijk te houden is het beter dit te scheiden. Net zo goed als een volledige update (PUT) en een gedeeltelijke update (PATCH) verschillende eisen stellen en totaal verschillende operaties zijn.

Daarnaast zijn van sommige resources alleen operaties voor het opvragen van lijsten beschikbaar en voor sommige andere resources ook operaties om details op te halen. Dat betekent dus dat de operatie raadplegen verschillende betekenissen kan hebben over resources heen. Dat maakt het minder overzichtelijk dan wanneer onderscheid gemaakt wordt tussen verschillende vormen van raadplegen (waar andere eisen aan gesteld worden).

> | *Raadplegen details*   |      |

Reviewer 5: Heb hier geen mening over omdat mij nog niet duidelijk is wie de doelgroep is (en wat die wil)

> | *Raadplegen details*   |      |

Author: Doelstelling van de documentatie is beschrijven wat er met de API kan zodat bijvoorbeeld een account-/productmanager of architect kan beoordelen of een API van toepassing is op een bepaald product. Dan is het belangrijk om te weten wat die API kan. Wanneer daar alleen globaal beschreven staat dat je kunt raadplegen weet je nog niets, zeker niet wanneer er verschillende manieren van raadplegen zijn. Vandaar de verwijzingen naar de verschillende vormen van raadplegen zoals die momenteel in de API ingebouwd zijn.

Alleen beschrijven dat CRUD operaties mogelijk zijn zegt helemaal niets, want dan zou ik vier regels verwachten waar niemand iets aan heeft. Documentatie jij voorstelt, terug naar alleen CRUD, heeft **geen** toegevoegde waarde, noch voor de niet-technische persoon die de OAS specificatie niet kan lezen noch voor de ontwikkelaar die dat wel kan.

Mijn voorstel is afgezien van de typefouten de documentatie voor nu te mergen en er later indien nodig een wijzigingsvoorstel te doen.

> | *Raadplegen details*   |      |

Reviewer: @michielverhoef, Consequentie van je voorstel is dat je elke operatie op een resource apart beschrijft. Anders is het soms wel en soms niet en dat is m.i. onduidelijk. Ben je dat van plan? M.i. ben je dan de API-specs aan het documenteren. M.i. moet dat in de API-spec zelf gebeuren.

> | *Raadplegen details*   |      |

Author: Volgens mij staan ook alle operaties beschreven en is dat ook de goede manier om te documenteren.  Van mij mag dat ook prima in de API spec zelf gebeuren maar dan is elke reden voor functionele documentatie weg.

Een hoogoververhaal waarmee je eigenlijk nog niets weet heeft geen toegevoegde waarde. Het is juist van belang te weten wat er aan functionaliteit (gerealiseerd in de verschillende operaties) mogelijk is.

Wat ook beschreven moet worden is de relatie van een component met andere componenten. Met name waar componenten elkaar moeten kunnen bereiken.  Voor de meeste componenten is dat wel duidelijk maar sommige relaties zijn dat mogelijk niet. Bijvoorbeeld dat een DRC een objectinformatieobject moet kunnen aanmaken in een ZRC. Het klinkt technisch maar is ook voor de (functionele) samenhang van belang om te weten.

| - voorwaarde   | De identificatie van de catalogus die opgevraagd wordt moet bekend zijn.  |

> | - gevolg       | Een specificieke catalogus wordt teruggegeven aan de consumer. |

Reviewer: ```suggestion
| - gevolg       | Een specificieke catalogus wordt teruggegeven aan de consumer. |
```

> | - gevolg       | Een specificieke catalogus wordt teruggegeven aan de consumer. |

Author: Ook hier geldt: Dat voorstel maakt het verhaal onduidelijk en daarmee slechter van kwaliteit. Ik zeg niet doen.

| Gegevens       | objecttype CATALOGUS inclusief een lijst met relaties naar daarin gedefinieerde ZAAKTYPEn.<br>Zie de volgende tabel voor de gegevens van deze resource                                 |
| Specificaties  | https://ref.tst.vng.cloud/ztc/api/v1/schema/#tag/catalogussen                                                                                                   |
| Samenhang      | - |
| Bijzonderheden | - |
        

### Resource: Zaaktypen

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aanleiding     | Een consumer wil zaaktypeinformatie opvragen uit de catalogus.            |

Author: Deze "lijst" was er blijkbaar tussendoor geschoten maar bij deze is ook deze tekst aangepast.

Je voorgestelde tekst kan ik dan weer niet plaatsen: een lijst van zaaktypen of gedetailleerde informatie van één specifiek zaaktype is toch allemaal zaaktype informatie? Bij de verschillende vormen van raadplegen kun je dan zien in welke formaten/structuren (lijst, details) die zaaktypeinformatie opgevraagd kan worden.

| *Toevoegen*    |      |
| - voorwaarde   | Nog niet beschikbaar.         |
| - gevolg       | Nog te bepalen.             |
| *Wijzigen*     |      |
| - voorwaarde   | Nog niet beschikbaar.         |
| - gevolg       | Nog te bepalen.             |
| *Verwijderen*  |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Raadplegen lijst*   |      |
| - voorwaarde   | Om zaaktypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het zaaktype is opgenomen. |
| - gevolg       | Een lijst met in de catalogus opgenomen zaaktypen wordt teruggegeven aan de consumer bij het opvragen van de lijst. |
| *Raadplegen details*   |      |

> | - voorwaarde   | Om specifieke zaaktypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het zaaktype is opgenomen. Tevens moet de identficatie van het zaaktype wat uitgevraagd wordt bekend zijn. |

Author: Om details op te vragen moet de identificatie van dat object bekend zijn. Welke dat is staat beschreven in de OAS.

De identficatie van de catalogus is waarschijnlijk onderdeel van de identificatie van het zaaktype maar dat weet ik niet zeker.
@sergei-maertens of @joeribekker , kunnen jullie hier iets over zeggen?

> | - voorwaarde   | Om specifieke zaaktypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het zaaktype is opgenomen. Tevens moet de identficatie van het zaaktype wat uitgevraagd wordt bekend zijn. |

Reviewer 3: > Om details op te vragen moet de identificatie van dat object bekend zijn. Welke dat is staat beschreven in de OAS.

❤️
> De identficatie van de catalogus is waarschijnlijk onderdeel van de identificatie van het zaaktype maar dat weet ik niet zeker.

Op dit moment is dat zo. Zie: https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/zaaktype_read

Je ziet hier bij parameters: `catalogus_uuid` en `uuid`. Beide zijn nodig om bij een Zaaktype uit te komen.

Één vraag van Arjan vind ik verder nog relevant:
> En als je de identificatie al weet, is dan de catalogus nog nodig?

Dat is een terechte vraag. In theorie niet, maar omdat via de API deze niet opvraagbaar is zonder catalogus identificatie, is deze nog wel nodig in de praktijk. Voor nu is het zo, dus kan vermeld worden in de documentatie.

> | - voorwaarde   | Om specifieke zaaktypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het zaaktype is opgenomen. Tevens moet de identficatie van het zaaktype wat uitgevraagd wordt bekend zijn. |

Reviewer 2: Op dit moment moet je inderdaad de UUID van het zaaktype kennen en het UUID van de catalogus om de resource-URL op te bouwen.

Wat in de praktijk vaak gebeurt, is dat deze URLs als instelling beschikbaar zijn bij een taakapplicatie, voorbeeld voor een MOR geef je een URL naar zaaktype in, voor Vergunningaanvraag geef je een andere URL in. Die URLs haal je uit de catalogus.

Wat je in de API url ziet is NIET de identificatie zoals die in RGBZ/ImZTC staat, dat is een arbitraire UUID als resource-identifier.

Afhankelijk van de noden (als taakapplicaties bijvoorbeeld eindgebruikers toelaten om een zaaktype identificatie in te voeren om daarmee te zoeken), kan er in de toekomst een filter parameter toegevoegd worden om zo de juiste zaaktype URL(s) te vinden. Dit bestaat nu echter niet.

In de demo applicatie was dit ook precies een database instelling: https://github.com/VNG-Realisatie/gemma-zaken-demo/blob/master/src/zac/demo/models.py#L21

> | - voorwaarde   | Om specifieke zaaktypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het zaaktype is opgenomen. Tevens moet de identficatie van het zaaktype wat uitgevraagd wordt bekend zijn. |

Reviewer: @sergei-maertens Ik kan het niet volgen. Die uuid is toch een unieke Id van een zaaktype? Daarmee vind je toch het zaaktype? Waar is die uuid van de catalogus dan nog voor nodig? Gaan we dit overal zo doen? Dus voor het lezen van een Statustype moet je de uuid van het statustype, die van het zaaktype en die van de catalogus meegeven, etcetera?

> | - voorwaarde   | Om specifieke zaaktypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het zaaktype is opgenomen. Tevens moet de identficatie van het zaaktype wat uitgevraagd wordt bekend zijn. |

Reviewer 2: @ArjanKloosterboer dit was per richtlijnen van de DSO, een zaaktype hoort bij een catalogus en moet daarom eronder genest worden. We zijn er zelf tijdens de eerste demo al achter gekomen dat dit niet heel erg handig is, maar goed, het zij nu zo en het wordt wel een keer opgepikt als de hoge-prio zaken weggewerkt zijn.

| - gevolg       | Een specificiek zaaktype wordt teruggegeven aan de consumer.         |

> | Gegevens       | Objecttype ZAAKTYPE inclusief STATUSTYPEn, EIGENSCHAPpen en ROLTYPEn<br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Reviewer: ```suggestion
| Gegevens       | Objecttype ZAAKTYPE inclusief relaties naar STATUSTYPEn, EIGENSCHAPpen en ROLTYPEn<br>Zie de volgende tabel voor de gegevens van deze resource                                 |
```

| *Toevoegen*    |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Wijzigen*     |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Verwijderen*  |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Raadplegen lijst*   |      |
| - voorwaarde   | Om besluittypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het besluittype is opgenomen. |
| - gevolg       | Een overzicht met besluittype informatie zoals opgenomen in de catalogus wordt teruggegeven.             |
| *Raadplegen details*   |      |
| - voorwaarde   | Om specifieke besluittypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het besluittype is opgenomen. Tevens moet de identficatie  van het besluittype wat uitgevraagd wordt bekend zijn. |
| - gevolg       | Besluittypeinformatie zoals opgenomen in de catalogus wordt teruggegeven.             |

> | Gegevens       | BESLUITTYPE <br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Author: De eerste opmerking begrijp ik niet, kun je hem verduidelijken? 
Waar moet "Objecttype" voor geplaatst worden?

> | Gegevens       | BESLUITTYPE <br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Reviewer: Een Besluittype maakt onderdeel uit van een Catalogus. Komt die relatie niet voor in de resource Besluittype?

| Specificaties  | https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/besluittype_read                                                                                |
| Samenhang      | - |
| Bijzonderheden | - |

### Resource: INFORMATIEOBJECTTYPE

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aanleiding     | Een consumer wil informatieobjecttypeinformatie van de informatieobjecten die voor kunnen komen bij het zaaktype zoals opgenomen in de catalogus van de organisatie opvragen.            |

Author: Klopt, ga ik aanpassen.

> | Aanleiding     | Een consumer wil informatieobjecttypeinformatie van de informatieobjecten zoals opgenomen in de catalogus van de organisatie opvragen.            |

Reviewer: ```suggestion
| Aanleiding     | Een consumer wil informatie opvragen van één of meer informatieobjecttypen die voorkomen in een zaaktypecatalogus van de organisatie.            |
```

Author: Voorstel overgenomen.

| *Toevoegen*    |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Wijzigen*     |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Verwijderen*  |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Raadplegen lijst*   |      |
| - voorwaarde   | Om informatieobjecttypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het besluittype is opgenomen. |

> | - gevolg       | Een overzicht van Informatieobjecttypeinformatie zoals opgenomen in de catalogus wordt teruggegeven.             |

Reviewer: Het is toch een overzicht van informatieobjecttypen?

Author: Klopt ook, pas ik aan.

Author: Is aangepast

| *Raadplegen details*   |      |

> | - voorwaarde   | Om details van informatieobjecttypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus het informatieobjecttype is opgenomen. Tevens moet de identficatie van het informatieobjecttype wat uitgevraagd wordt bekend zijn. |

Reviewer: Zie opmerking hiervoor over de identificatie.

Author: Zie mijn antwoord op de opmerking hiervoor

| - gevolg       | Informatieobjecttypeinformatie zoals opgenomen in de catalogus wordt teruggegeven.             |

> | Gegevens       | INFORMATIEOBJECTTYPE <br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Reviewer: 'Objecttype ...' er voor. Geen relaties? Hoe zit het met die naar Catalogus? En naar Besluittype? En naar Zaaktype?

Author: Die staan momenteel niet in de API als ik het goed zie.

Reviewer: Hoe kan dat nou? Snap ik niet.

Reviewer: @sergei-maertens Hoe zit dat?

| Specificaties  | https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/informatieobjecttype_read                                                                                |
| Samenhang      | - |
| Bijzonderheden | - |

> ### Resource: Eigenschapen

Reviewer 5: Eigenschapen -> Eigenschappen

Author: Woeps, fixed

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aanleiding     | Een consumer wil alle eigenschappen van een zaaktype zoals opgenomen in de catalogus van de organisatie opvragen.             |

Reviewer: Lijkt in tegenspraak met wat bij de tweede raadpleegoperatie staat.

Author: Kopieer foutje

Author: Is aangepast

Reviewer: ```suggestion
| Aanleiding     | Een consumer wil een eigenschap of een overzicht van alle eigenschappen opvragen van een zaaktype zoals opgenomen in de catalogus van de organisatie.             |
```

Author: Dat voorstel maakt het verhaal onduidelijk en daarmee slechter van kwaliteit. Ik zeg niet doen.

Reviewer: @michielverhoef, welke mogelijkheden biedt de API qua Eigenschappen? Alleen maar opvragen van alle eigenschappen van een zaaktype? Dus niet een enkele (één) eigenschap van een zaaktype opvragen? In jouw beschrijving staat alleen het eerste. Dan zou het tweede dus niet ondersteuntd worden door de API. Is dat juist?

Author: Ik verwacht dat ook een deel van de eigenschappen van een objectype opgevraagd kan worden. Maar daar kunnen Sergei en/of Joeri beter antwoord op geven.

De aanleiding staat momenteel als volgt omschreven:
"Een consumer wil zaaktypeinformatie opvragen uit de catalogus."

Volgens mij dekt dat de lading voor zowel een lijst van zaaktypen als zaakdetails (alle details of een beperkt aantal).

Reviewer: > De aanleiding staat momenteel als volgt omschreven: "Een consumer wil zaaktypeinformatie opvragen uit de catalogus." Volgens mij dekt dat de lading voor zowel een lijst van zaaktypen als zaakdetails (alle details of een beperkt aantal).

Het gaat hier over het objecttype EIGENSCHAP met een n-1-relatie naar ZAAKTYPE. Dus niet over zaaktypen noch over zaakdetails. Een 'list' geeft m.i. een overzicht van alle eigenschappen bij een zaaktype. Een 'read' geeft m.i. alle informatie over één eigenschap bij een zaaktype. 
Je aanleidingtekst klopt dan niet.

Reviewer 2: Op dit moment kan je niet filteren/limiteren op 1 eigenschap, dus je krijgt of de hele lijst van de eigenschappen, of je vraagt de details op voor 1 enkele eigenschap.

| *Toevoegen*    |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Wijzigen*     |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Verwijderen*  |      |
| - voorwaarde   | Nog niet beschikbaar.  |
| - gevolg       | Nog te bepalen.             |
| *Raadplegen lijst*   |      |
| - voorwaarde   | Om eigenschapinformatie op te kunnen vragen moet de consumer weten in welke catalogus en bij welk zaaktype de eigenschap is opgenomen. |

> | - gevolg       | Een overzicht van eigenschapinformatie van het zaaktype wordt teruggegeven.            |

Reviewer: Is toch een overzicht van eigenschappen?

Author: Mja, meer van eigenschap-typen zou ik zeggen. De eigenschap heeft een waarde, dat heeft de typedefnitie niet.

Author: Ik resolve de conversation

Reviewer: Is dit consistent met andere vermeldingen van het gevolg van Raadplegen lijst?

| *Raadplegen details*   |      |

> | - voorwaarde   | Om details van eigenschapinformatie op te kunnen vragen moet de consumer weten in welke catalogus en bij welk zaaktype de eigenschap is opgenomen. Tevens moet de identficatie van de eigenschap welke uitgevraagd wordt bekend zijn. |

Reviewer: Zie eerdere opmerking over de identificatie.

Author: Ook hiervoor gelden de vragen die ik bij die opmerking stelde.

| - gevolg       | Detailinformatie van de eigenschap wordt teruggegeven.            |

> | Gegevens       | EIGENSCHAP <br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Reviewer: 'Objecttype ...' er voor. En relatie naar Zaaktype?

Author: Die moet er nog bij inderdaad

Author: gedaan

| Specificaties  | https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/eigenschap_read                                                                                |

> | Samenhang      | Een relevant inhoudelijk gegeven dat bj ZAAKen van dit ZAAKTYPE geregistreerd moet kunnen worden en geen standaard kenmerk is van een ZAAK. |

Reviewer: Wat is de reden om dit te vermelden, en idem hier? Het lijkt mij gekopieerd uit het ImZTC. Dat lijkt me niet zinvol, niet voor niets wordt er naar het ImZTC en het objecttype Eigenschap verwezen. Houd het hier bij functionaliteit.

Author: Ok. In dat geval zie ik nog weinig inhoud voor het aspect samenhang en zouden we dat kunnen laten vervallen.

Reviewer: Eerst maar 's de documentatie voor enkele API's uitwerken. En dan 's evalueren en waar nodig bijstellen.

| Bijzonderheden | - |
      
### Resource: Roltypen

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aanleiding     | Een consumer wil alle roltypen van een zaaktype zoals opgenomen in de catalogus van de organisatie opvragen.            |

Reviewer: ```suggestion
| Aanleiding     | Een consumer wil informatie opvragen van een specifiek roltype of van alle roltypen van een zaaktype zoals opgenomen in de catalogus van de organisatie.            |
```

| *Toevoegen*    | Nog niet beschikbaar.     |
| - voorwaarde   | Nog te bepalen.  |
| - gevolg       | Nog te bepalen.             |
| *Wijzigen*     | Nog niet beschikbaar.     |
| - voorwaarde   | Nog te bepalen.  |
| - gevolg       | Nog te bepalen.             |
| *Verwijderen*  | Nog niet beschikbaar.     |
| - voorwaarde   | Nog te bepalen.  |
| - gevolg       | Nog te bepalen.             |
| *Raadplegen lijst*   |      |
| - voorwaarde   | Om roltypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus en bij welk zaaktype het roltype is opgenomen. |
| - gevolg       | Een lijst met roltypeinformatie behorend bij het zaaktype wordt teruggegeven aan de consumer.             |
| *Raadplegen details*   |      |
| - voorwaarde   | Om roltypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus en bij welk zaaktype het roltype is opgenomen. Tevens moet de identficatie van het roltype wat uitgevraagd wordt bekend zijn. |
| - gevolg       | Roltypeinformatie wordt teruggegeven aan de consumer.             |

> | Gegevens       | Objecttype ROLTYPE <br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Reviewer: Geen relatie naar het Zaaktype waar het bijhoort?

Author: Correct, is toegevoegd.

| Specificaties  | https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/roltype_read                                                                                                   |
| Samenhang      | - |
| Bijzonderheden | - |

       
### Resource: Statustypen

| **Aspect**     | **Beschrijving**                                                                                                                                                                 |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aanleiding     | Een consumer wil statustypeinformatie van een ZAAKTYPE zoals opgenomen in de CATALOGUS van de organisatie opvragen.           |

Reviewer: ```suggestion
| Aanleiding     | Een consumer wil informatie opvragen van een specifiek statustype of van alle statustypen van een zaaktype zoals opgenomen in de catalogus van de organisatie.           |
```

Author: Overgenomen

| *Toevoegen*    | Nog niet beschikbaar.     |
| - voorwaarde   | Nog te bepalen.  |
| - gevolg       | Nog te bepalen.             |
| *Wijzigen*     | Nog niet beschikbaar.     |
| - voorwaarde   | Nog te bepalen.  |
| - gevolg       | Nog te bepalen.             |
| *Verwijderen*  | Nog niet beschikbaar.     |
| - voorwaarde   | Nog te bepalen.  |
| - gevolg       | Nog te bepalen.             |
| *Raadplegen lijst*   |      |
| - voorwaarde   | Om statustypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus en bij welk zaaktype het statustype is opgenomen. |
| - gevolg       | Een lijst met statustypeinformatie wordt teruggegeven aan de consumer.             |
| *Raadplegen details*   |      |
| - voorwaarde   | Om statustypeinformatie op te kunnen vragen moet de consumer weten in welke catalogus en bij welk zaaktype het statustype is opgenomen. Tevens moet de identficatie van het statustype wat uitgevraagd wordt bekend zijn. |
| - gevolg       | Statustypeinformatie wordt teruggegeven aan de consumer.             |

> | Gegevens       | Objecttype STATUSTYPE <br>Zie de volgende tabel voor de gegevens van deze resource                                 |

Reviewer: Geen relatie naar het Zaaktype waar het bijhoort?

Author: Op dit moment staat die relatie niet opgenomen in de API dus ook niet beschreven. Zie ook de uitleg van Sergei.

---
title: "Zaaktypen-API"
date: '19-01-2019'
---

# ZAAKTYPEN API

FUNCTIONELE BESCHRIJVING

Hieronder lichten we de gebruiksmogelijkheden van deze API toe. Voor de
werking van deze API verwijzen wij u naar de OAS-specificatie (zie
‘link’ hieronder).

<table>
<tbody>
<tr class="odd">
<td><strong>Aspect</strong></td>
<td><strong>Beschrijving</strong></td>
</tr>
<tr class="even">
<td>Doel</td>

> <td>Het kunnen aanspreken van een voorziening voor het onderhouden en raadplegen van zaken inclusief bijbehorende statussen, rollen en relaties naar betrokkenen, objecten, zaaktypen en statustypen.</td>

Reviewer 2: Dit is een copy-paste van zaken-api documentatie? Moet m.i. focussen op zaaktypen/informatieobjecttypen/resultaattypen/objecttypen etc.

Author: Oeps.. Omschrijving is aangepast.

</tr>
<tr class="odd">
<td>Bijzonderheden</td>
<td><ul>
<li>Het aanmaken van Informatieobjectype wordt nog niet ondersteund.</li>
<li>Het verwijderen van Informatieobjectype wordt nog niet ondersteund.</li>
<li>Het aanpassen van Informatieobjectype wordt nog niet ondersteund.</li>
</ul></td>
</tr>
</tbody>
</table>

### Resource: Eigenschappen

<table>
<tbody>
<tr class="odd">
<td><strong>Aspect</strong></td>
<td><strong>Beschrijving</strong></td>
</tr>
<tr class="even">
<td>Doel</td>
<td><ul>
        <li><a href="https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/eigenschap_list">Opvragen</a> van een overzicht van alle beschikbare Eigenschappen zoals gedefinieerd bij een Zaaktype, met de gegevens per Eigenschap.</li>
    <li><a href="https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/eigenschap_read">Opvragen</a> van de gegevens van een specifieke Eigenschap.</li>
</ul></td>
</tr>
<tr class="odd">
<td>Gegevens</td>
<td>Objecttype EIGENSCHAP, inclusief relatie naar ZAAKTYPE. https://www.gemmaonline.nl/index.php/Imztc_2.2/doc/objecttype/eigenschap</td>
</tr>
<tr class="even">
<td>OAS-specificaties</td>

> </tr>

Reviewer 2: deze closing tag zorgt ervoor dat de tabel niet helemaal lekker gaat

Author: closing tag verwijderd

> </tr>

Reviewer 2: ```suggestion
```

Author: overgenomen

> <td><a href="https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/eigenschap_list">https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/eigenschap_list</a></td>

Reviewer 2: ```suggestion
<td><a href="https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/eigenschap_list">https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/eigenschap_list</a></td></tr>
```

Author: overgenomen

<tr class="odd">
<td>Bijzonderheden</td>
<td><ul>
<li>Het aanmaken van Eigenschappen wordt nog niet ondersteund.</li>
<li>Het verwijderen van Eigenschappen wordt nog niet ondersteund.</li>
<li>Het aanpassen van Eigenschappen wordt nog niet ondersteund.</li>
</ul></td>
</tr>
</tbody>
</table>

### Resource: Roltypen

<table>
<tbody>
<tr class="odd">
<td><strong>Aspect</strong></td>
<td><strong>Beschrijving</strong></td>
</tr>
<tr class="even">
<td>Doel</td>
<td><ul>
     <li><a href="https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/roltype_list">Opvragen</a> van een overzicht van alle beschikbare Roltypen.</li>
    <li><a href="https://ref.tst.vng.cloud/ztc/api/v1/schema/#operation/roltype_read">Opvragen</a> van de gegevens van een specifiek Roltype.</li>
</ul></td>
</tr>
<tr class="odd">
<td>Gegevens</td>

> <td>Objecttype ROLTYPE, inclusief relatie naar ZAAKTYPE en mogelijkeBETROKKENEN. https://www.gemmaonline.nl/index.php/Imztc_2.2/doc/objecttype/roltype</td>

Reviewer 2: ```suggestion
<td>Objecttype ROLTYPE, inclusief relatie naar ZAAKTYPE en mogelijke BETROKKENEN. https://www.gemmaonline.nl/index.php/Imztc_2.2/doc/objecttype/roltype</td>
```

Author: overgenomen
